src/charm.py

```diff
@@ -1129,12 +1129,16 @@ def _get_loadbalancer_status(namespace: str, service_name: str) -> Optional[str]
     client = Client()  # type: ignore
     traefik_service = client.get(Service, name=service_name, namespace=namespace)
 
-    if status := traefik_service.status:  # type: ignore
-        if load_balancer_status := status.loadBalancer:
-            if ingress_addresses := load_balancer_status.ingress:
-                if ingress_address := ingress_addresses[0]:
-                    return ingress_address.hostname or ingress_address.ip
-    return None
+    if not (status := traefik_service.status):  # type: ignore
+        return None
+    if not (load_balancer_status := status.loadBalancer):
+        return None
+    if not (ingress_addresses := load_balancer_status.ingress):
+        return None
+    if not (ingress_address := ingress_addresses[0]):
+        return None
+
+    return ingress_address.ip
```

Comment:
I'm guessing that the only thing we'd be losing here is when `ingress_address.hostname` evaluates to something sensible that was not set as a config option. Seems reasonable.

Perhaps just add a comment saying we intentionally disregard it?

Reply:
done